### DISCUSSION TITLE
Shark Vacuum Error Codes and Error messages

### DISCUSSION THREAD

**Discussion by 400HPMustang** [sharkiqlibs/sharkiq#46] [Ideas]
We don't currently have a list of error codes or error messages that occur when the robot encounters a problem. Having this would be useful for building automations or other notifications or what have you.

Starting a collection here from the user base, if we keep collecting these we might have an entire library to use. . One of my robots currently spit out:

```
Last error code: 6

Last error message: Bumper is stuck
```

**Comment by ljonesfl** [comment]
Brushroll stuck, 4

**Comment by Slivacki** [comment]
I was able to get an automation to detect an error but you need all these names to trigger it. Here is list I came up with once I figured that out.

Shark unit model
AV251WAXUS

1
2 Side brush is stuck 
3
4 Brushroll stuck 
5 Side wheel is stuck (2) 
6 Bumper is stuck
7 Cliff sensor is blocked 
8
9 No Dustbin
10 Fall sensor is blocked

**Reply by Slivacki** [reply]
Looking around in the integration I was able to find the list of error codes the integration uses. 

I have one that wasn't on the list 30: "Unknown error (30)" that my unit kicked out.

ERROR_MESSAGES 
    1: "Side wheel is stuck",
    2: "Side brush is stuck",
    3: "Suction motor failed",
    4: "Brushroll stuck",
    5: "Side wheel is stuck (2)",
    6: "Bumper is stuck",
    7: "Cliff sensor is blocked",
    8: "Battery power is low",
    9: "No Dustbin",
    10: "Fall sensor is blocked",
    11: "Front wheel is stuck",
    13: "Switched off",
    14: "Magnetic strip error",
    16: "Top bumper is stuck",
    18: "Wheel encoder error",
    40: "Dustbin is blocked",
